src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -69,27 +67,33 @@ public Path which(String fn) {
 
     @Override
     public String[] getCommand() {
-        String[] decorated = cmds;
+        String[] decorated = Arrays.copyOf(cmds, cmds.length);
+        for (int i = 0; i < decorated.length; i++) {
+            final String arg = decorated[i];
+            // Space and \t require quoting otherwise will be split to more than one arg
+            if (!isQuoted(arg) && arg.matches(".*[ \t].*")) {
+                decorated[i] = "\"" + arg + "\"";
+            }
+        }
         if (shellDecorator != null) {
             decorated = shellDecorator.decorate(decorated);
         }
-        // First item in the command is the executable. If it's given as absolute path, add quotes around it
-        // in case the path contains space which will break the whole command line
-        // See security remarks:
-        // https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createprocesswithlogonw
-        if (isAbsolutePath(decorated[0])) {
-            decorated[0] = String.format("\"%s\"", decorated[0]);
-        }
         return decorated;
     }
 
     @Override
-    protected Process createProcess() {
-        WindowsRunasProcess winProcess = new WindowsRunasProcess(null, userDecorator.getUser());
-        winProcess.setLpEnvironment(computeEnvironmentBlock());
-        winProcess.setLpCurrentDirectory(dir.getAbsolutePath());
-        winProcess.start(String.join(" ", getCommand()));
-        return winProcess;
+    protected Process createProcess() throws IOException {
+        if (needToSwitchUser()) {
+            WindowsRunasProcess winProcess = new WindowsRunasProcess(LOCAL_DOMAIN, userDecorator.getUser());
```

Comment:
what do we do if we can't find the password for the user? Why are we forcing local domain? Why about networked accounts?

Reply:
Need to do verification in RunWithGenerator. Check like user exists and we have the credential